src/app/content/highlights/hooks/loadMore.ts

```diff
@@ -1,72 +1,52 @@
-
-const incrementPage = (pagination: NonNullable<SummaryHighlightsPagination>) =>
-  ({...pagination, page: pagination.page + 1});
-
-const getNewSources = (state: AppState, omitSources: string[]) => {
-  const book = bookSelector(state);
-  const remainingCounts = omit(omitSources, select.filteredCountsPerPage(state));
-  return book ? getNextPageSources(remainingCounts, book.tree, summaryPageSize) : [];
-};
-
-const loadUntilPageSize = async({
+import {
+  fetchHighlightsForSource,
+  formatReceivedHighlights,
+  getNewSources,
+  incrementPage,
+} from './utils';
+
+export const loadUntilPageSize = async({
   previousPagination,
   ...args
 }: {
-  previousPagination: SummaryHighlightsPagination,
-  getState: Store['getState'],
+  previousPagination: SummaryHighlightsPagination, getState: Store['getState'],
   highlightClient: AppServices['highlightClient'],
   highlights?: Highlight[]
-  sourcesFetched: string[]
+  sourcesFetched: string[],
+  pageSize?: number,
 }): Promise<{pagination: SummaryHighlightsPagination, highlights: Highlight[]}> => {
   const state = args.getState();
   const book = bookSelector(state);
   const {colors} = select.summaryFilters(state);
-  const {page, sourceIds} = previousPagination
+  const {page, sourceIds, perPage} = previousPagination
     ? incrementPage(previousPagination)
-    : {sourceIds: getNewSources(state, args.sourcesFetched), page: 1};
+    : {
+      page: 1,
+      perPage: args.pageSize || maxHighlightsPerPage,
+      sourceIds: getNewSources(state, args.sourcesFetched, args.pageSize),
+    };
 
   if (!book || sourceIds.length === 0) {
     return {pagination: null, highlights: args.highlights || []};
   }
 
-  const highlightsResponse = await args.highlightClient.getHighlights({
+  const {highlights, pagination} = await fetchHighlightsForSource({
+    book,
     colors: colors as unknown as GetHighlightsColorsEnum[],
-    page,
-    perPage: summaryPageSize,
-    scopeId: book.id,
-    sourceIds,
-    sourceType: GetHighlightsSourceTypeEnum.OpenstaxPage,
+    highlightClient: args.highlightClient,
+    pagination: {page, sourceIds, perPage},
+    prevHighlights: args.highlights,
   });
 
-  // TODO - change swagger so none of this is nullable
-  const data = assertDefined(highlightsResponse.data, 'response from highlights api is invalid');
-  const meta = assertDefined(highlightsResponse.meta, 'response from highlights api is invalid');
-  const perPage = assertDefined(meta.perPage, 'response from highlights api is invalid');
-  const totalCount = assertDefined(meta.totalCount, 'response from highlights api is invalid');
-
-  const loadedResults = (page - 1) * perPage + data.length;
-  const pagination = loadedResults < totalCount
-    ? {sourceIds, page}
-    : null;
-
-  const highlights = args.highlights
-    ? [...args.highlights, ...data]
-    : data
-  ;
-
-  if (highlights.length < summaryPageSize) {
+  if (highlights.length < summaryPageSize || !args.pageSize) {
```

Comment:
I agree with replacing `summaryPageSize` with `perPage` but i think `!args.pageSize` has to stay. Because usually after requesting to print highlights, there will be a pagination object coming from previous scroll event which has to be 'used up' before continuing with `maxHighlightsApiPageSize`.